pennylane/tape/qnode.py

```diff
@@ -140,13 +140,16 @@ def __init__(self, func, device, interface="autograd", diff_method="best", **dif
             )
 
         self.func = func
-        self.device = device
+        self.original_device = device
         self.qtape = None
         self.qfunc_output = None
+        self.diff_method = diff_method

```

Comment:
Ah, this is a subtlety that confused me as well. The issue is that the `diff_method` argument for the QNode, and the `diff_method` argument for the `Tape.jacobian()`:

* QNode: accepts `best`, `parameter-shift`, `finite-diff`, `device`, `backprop`
* Tape: accepts `best`, `numeric`, `analytic`, `numeric` 

🙁 

So we need to store both the QNode argument _and_ the `tape.jacobian` argument for further use. It's a bit annoying.

Here, `self.diff_method` stores the QNode argument (since I think that makes more sense), whereas `self.diff_options` stores the arguments given to `Tape.jacobian`.

```suggestion
        # store the user-specified differentiation method
        self.diff_method = diff_method
```